### ISSUE TITLE
willing to contribute

### ISSUE THREAD

**Issue by dk1978** [mstaz/android_device_fairphone_fp3#1]
Hi,
I have a Fairphone 3 myself and am willing to contribute to a LineageOS built - and continue with Halium and UBports later. Let me know if I can help with anything.
BR,
dk

**Comment by mstaz** [comment]
Hi dk,
I'd appreciate any help.
Please feel free to start with my config and get things running. At the moment we're quite at the beginning. However we can try to go step by step and get the things running :smile:
Also I don't have any contact to the LOS developers. Maybe we could also contact them to get some help from them as well.
BR,
Max

**Comment by dk1978** [comment]
Hi Max,

I am writing my reply here again, because my first reply through direct email probably ended nowhere. Thanks for the feedback. I have forked your device repo.

My first suggestion: on mstaz/android_device_fairphone_fp3 create a branch "lineage-16.0" and make it default. This is what the repo and build scripts seem to expect.

I have created some XML code that goes into .repo/local_manifests/roomservice.xml and should allow any newcomers a clean "repo sync" command. I will document the full procedure on my fork's README.md later today.

Right now I am looking at some strange warnings that occurred while configuring the kernel. The build warnings go something like this:
`rivers/input/touchscreen/hxchipset83112b/Kconfig:5:warning: ignoring unsupported character`
It turns out that the kernel sub-folder drivers/input/touchscreen/hxchipset83112b contains a lot of weird .i files, and it looks like the directory content was pulled off an NTFS or FAT* filesystem by either Arima or Fairphone. The DOS line endings in the Kconfig files are what triggers the warnings. I am considering a clean-up of this directory to eliminate the warnings.

Regards,
Daniel

**Comment by Daniel6801** [comment]
Hi Max, hi Daniel,

I am looking forward to buy the fairphone, but will only buy it, if 
there is a chance that LineageOS will run on it. I am very happy to see 
this project and if I can help, please let me know.

If you like, we may use a XMPP group chat (e. g. Conversations App on 
Android) for ease of communication as github offers not much options to 
communicate... I am sorry, the very common WhatsApp is no option for me 
(I don't install "Spyware" on my phone...).

I am a Java developer, but I have no experience and knowledge with 
Android. Currently I try to get the sources compiled, but end up with error:

[1037/1037] including vendor/qcom/opensource/dataservices/Android.mk ...
device/fairphone/fp3/libshims/Android.mk: error: "libqsap_shim 
(SHARED_LIBRARIES android-arm64) missing libqsap_sdk (SHARED_LIBRARIES 
android-arm64)"
You can set ALLOW_MISSING_DEPENDENCIES=true in your environment if this 
is intentional, but that may defer real problems until later in the build.
device/fairphone/fp3/libshims/Android.mk: error: "libqsap_shim 
(SHARED_LIBRARIES android-arm) missing libqsap_sdk (SHARED_LIBRARIES 
android-arm)"
You can set ALLOW_MISSING_DEPENDENCIES=true in your environment if this 
is intentional, but that may defer real problems until later in the build.
build/make/core/main.mk:844: error: exiting from previous errors.
22:42:42 ckati failed with: exit status 1

I tried with "ALLOW_MISSING_DEPENDENCIES=true", but then a similar error 
occurs a bit later in the build.

I will try it again with the changes in the fork from dk1978 in the next 
days (I have no more time today for further retries...).

Regards

Daniel

**Comment by mstaz** [comment]
Hi Daniel,

Great to hear.
Yes, I definitely should work with clean branches :)
Regarding the errors you can also check the additional changes from k4y0z and z3ntu. Maybe they already fixed these.
I'll check for that tomorrow.
BR,
Max


Am 4. März 2020 16:00:58 MEZ schrieb dk1978 <notifications@github.com>:

**Comment by mstaz** [comment]
Hi Daniel,
Great that you participate porting LOS.
These errors should not happen. I try to check for it tomorrow.
Yes, sure. A group chat would be nice. Maybe there are also "official" LOS channels available we could use.
BR,
Max


Am 4. März 2020 22:58:22 MEZ schrieb Daniel6801 <notifications@github.com>:

**Comment by Daniel6801** [comment]
The compile runs without error now with the latest changes and with stock firmware 110. Thank's a lot!

It took me some time to find out how to extract the downloaded firmware dump which did not contain the *.img files. Just some hint's how to extract these *.br and *.dat files: (@mstaz, dk1978: maybe this could be added into the README.md?):

The *.br files can be extracted with brotli.

On Linux (debian) install brotli via (I assume similar tools exists for MS-Windows):
  sudo apt-get instal brotli
Extract *.br files with:
  brotli -d *.br

Afterwards the *.dat files can be converted into the *.img files with the "Extraction and repack tool":
  git clone https://github.com/iykequame/android_system_extraction_and_repack_tool.git

Extract with python script sdat2img.py:
  ./tools/sdat2img.py system.transfer.list  system.new.dat  system.img
  ./tools/sdat2img.py vendor.transfer.list  vendor.new.dat  vendor.img
  ./tools/sdat2img.py product.transfer.list product.new.dat product.img

**Comment by mstaz** [comment]
Yes, needs to be extracted with brotli first. I'll add your description to the readme.
LOS already comes with the sdat2img tool. It should be here: `vendor/lineage/build/tools/sdat2img.py`
